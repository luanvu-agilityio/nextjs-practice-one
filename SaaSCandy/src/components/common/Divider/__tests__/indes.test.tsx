import { render, screen } from '@testing-library/react';
import { Divider } from '..';

describe('Divider', () => {
  it('renders match snapshot', () => {
    const { container } = render(<Divider />);
    expect(container).toMatchSnapshot();
  });

  it('renders with default props', () => {
    render(<Divider />);

    const text = screen.getByText('OR');
    expect(text).toBeInTheDocument();
    expect(text).toHaveClass('text-sm', 'text-gray-500', 'font-medium');
  });

  it('renders with custom text', () => {
    render(<Divider text='AND' />);

    const text = screen.getByText('AND');
    expect(text).toBeInTheDocument();
  });

  it('applies custom className', () => {
    const { container } = render(<Divider className='custom-class' />);

    const divider = container.firstChild as HTMLElement;
    expect(divider).toHaveClass('custom-class');
  });
});
