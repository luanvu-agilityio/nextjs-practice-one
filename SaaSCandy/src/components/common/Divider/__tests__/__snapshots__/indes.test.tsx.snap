// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Divider > renders match snapshot 1`] = `
<div>
  <div
    class="relative flex items-center justify-center "
  >
    <div
      class="absolute inset-0 flex items-center"
    >
      <div
        class="w-full border-t border-gray-200"
      />
    </div>
    <div
      class="relative bg-white px-4"
    >
      <span
        class="text-sm text-gray-500 font-medium"
      >
        OR
      </span>
    </div>
  </div>
</div>
`;

exports[`Divider renders match snapshot 1`] = `
<div>
  <div
    class="relative flex items-center justify-center "
  >
    <div
      class="absolute inset-0 flex items-center"
    >
      <div
        class="w-full border-t border-gray-200"
      />
    </div>
    <div
      class="relative bg-white px-4"
    >
      <span
        class="text-sm text-gray-500 font-medium"
      >
        OR
      </span>
    </div>
  </div>
</div>
`;
